pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/windowing/Pair.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.functions.windowing;
+
+import java.io.Serializable;
+
+/**
+ * A pair of values.
+ *
+ * @param <T1> the type of the first value
+ * @param <T2> the type of the second value
+ */
+public final class Pair<T1, T2> implements Serializable {
```

Comment:
There's already a [`Pair`](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/tuple/Pair.html) class used elsewhere in the Pulsar codebase. Probably best to use that rather than creating a parallel, homespun variant.

Reply:
I implore you to use an existing implementation if at all possible.